samples: nfc: writable_ndef_msg: Select FLASH_MAP when ZMS is enabled

doki-nordic opened the pull request:
The sample in Kconfig sets default ZMS to 'y', but CONFIG_FLASH_MAP must also be set in this case.

Jira: NCSDK-35447
Jira: NCSDK-35446

tomchy committed:
applications: nrf_desktop: Do not unset nRF5 bootloader

Remove override for (now promptless) Kconfig.
Remove the override to ROM_START_OFFSET, which is already changed if the
partition manager is enabled.

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

bjarki-andreasen committed:
drivers: mspi: remove redundant gpd handling

Manual handling of global power domains using gpd has been
made redundant by power domain and pinctrl refactoring, and gpd
has thus been removed. Remove gpd handling from mspi driver, as
this is automatically done by pinctrl_apply_state() now.

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

bjarki-andreasen committed:
samples: smp_svr: extend mcuboot_merged_slot.overlay

The board overlay in bootloader/mcuboot for the
nrf54h20dk/nrf54h20/cpuapp platform is required, but ignored as
FILE_SUFFIX="merged_slot" matches the mcuboot_merged_slot.overlay
in this sample instead.

Copy the missing contents of the board overlay from
bootloader/mcuboot to mcuboot_merged_slot.overlay

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

bjarki-andreasen committed:
nrf_compress: decompression: mcuboot_update: adjust nrf54h20dk pd

Adjust nrf54h20dk/nrf54h20/cpuapp overlay and conf to include
required props from the overlay in bootloader/mcuboot. Specifically
disabling power domains.

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

bjarki-andreasen committed:
benchmarks: multicore: idle: select PM_DEVICE_RUNTIME

The global power domains require PM_DEVICE to be power managed.
Enable PM_DEVICE, and unselect the default, optional, not required
PM_DEVICE_RUNTIME_ASYNC feature.

Signed-off-by: Bjarki Arge Andreasen <bjarki.andreasen@nordicsemi.no>

jukkar committed:
manifest: Upmerge sdk-zephyr, sdk-mbedtls and sdk-mcuboot

Regular upmerge.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

cvinayak committed:
boards: Remove usage of deprecated BT_CTLR Kconfig option

Use HAS_BT_CTLR instead of the deprecated BT_CTLR.

Relates to zephyrproject-rtos/zephyr#94287

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

ArekBalysNordic committed:
samples: openthread: Switch to new USB driver

In Zephyr the current USB driver used in CLI and coprocessor
samples has been deprecated, and we need to switch to the new
usbd one.

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>

rlubos committed:
samples: net: mqtt: Align with SMF API changes

The state_execution() function signature changed.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

juhaylinen committed:
tests: net: lwm2m: Align with API changes

The clear_attrs() function signature changed.

Signed-off-by: Juha Ylinen <juha.ylinen@nordicsemi.no>

ArekBalysNordic committed:
samples: openthread: Align coprocessor HCI to new USB stack

We need to provide some alignments to the Coprocessor sample after
switching to the new USB stack.

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>

tokangas committed:
treewide: Remove use of CONFIG_DEBUG_INFO

CONFIG_DEBUG_INFO has been renamed to CONFIG_X86_DEBUG_INFO. It is an
X86 specific option and should never have been used.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

jfischer-no committed:
samples: bluetooth: llpm: remove nrf52840dongle USB configuration

nrf52840dongle uses CDC ACM serial backend by default.
Disable logging as is not used in the sample.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

kl-cruz committed:
samples: i2c: Add missing Kconfig to rtio_loopback sample

Add Kconfig with sourcing main Kconfig file.

Signed-off-by: Karol Lasończyk <karol.lasonczyk@nordicsemi.no>

jfischer-no committed:
samples: bluetooth: peripheral_uart: use CDC ACM serial backend

Use CDC ACM serial backend from the new USB device support.

There seems to be an issue with the sample that it does not always work
and crashes somewhere between UART configuration and Bluetooth
initialization.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
samples: nrf5340: remote_shell: use CDC ACM serial backend

Use CDC ACM serial backend from the new USB device support.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

ppryga-nordic committed:
bluetooth: In hci_internal_msg_get add msg_type_out cast to avoid warn

In the hci_internal_msg_get() there is a call to sdc_hci_get()
that second parameter is uint8_t pointer.
The provided msg_type_out argument is an enum sdc_hci_msg_type_t
pointer that must fit into the uint8_t. Durign build compiler
generates an error about incompatible pointer types.

The commit adds a cast to uint8_t to fix the compilation error.

Note: the enum sdc_hci_msg_type_t is defined in SoftDevice
Controller headers. It is SoftDevice responsibility to make
sure the enum doesn't contain values that don't fit into
uint8_t.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

simensrostad committed:
modules: memfault-firmware-sdk: Fix SMF handling

SMF run functions now returns SMF_EVENT_HANDLED or SMF_EVENT_PROPAGATE.
Fix this for all occurances in memfault_lte_coredump.c

Signed-off-by: Simen S. Røstad <simen.rostad@nordicsemi.no>

tokangas committed:
boards: nordic: thingy91x: Remove use of deprecated Kconfig option

Removed use of deprecated NFCT_PINS_AS_GPIOS Kconfig option and
added the NFCT pin configuration into devicetree instead.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

tokangas committed:
applications: serial_lte_modem: Remove undefined Kconfig options

Removed a few Kconfig options which were in comments, because those
no longer exist and caused an error.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

ppryga-nordic committed:
bluetooth: cs_de: Fix after combined main and sub mode CS steps

There was a change in sdk-zephyr Bluetooth host implementation
that combined main and sub CS mode steps.

Not all combinations of main mode and sub mode are valid, so to make it
harder to select incorrect parameters, the main mode and sub mode
enums are combined into one.

With that change names of some CS related enums were changed.
The commit aligns sdk-nrf to use the new names.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

tokangas committed:
treewide: Remove TinyCrypt Kconfig options from conf files

Removed commented out TinyCrypt related Kconfig options from
conf files. The Kconfig options no longer exist and even
if the options are in comments, those still cause an error.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

tokangas committed:
samples: wifi: shell: Remove CONFIG_NFCT_PINS_AS_GPIOS from overlay

The CONFIG_NFCT_PINS_AS_GPIOS Kconfig option no longer exists and the
configuration is done through devicetree. The feature is enabled by
default by the board, so there's no need to enable it.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

rado17 committed:
samples: wifi: softap: Fix twister errors

Correct the return type for `.run` functions of SMF. Enable
SETTINGS which is required for wifi_credentials support.

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>

ArekBalysNordic committed:
applications: matter: Fix Weather Station USB stack

Switched to the new USB stack.

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>

katgiadla committed:
quarantine: Add overflows to quarantine

Test configurations will added to quarantine due to
RAM overflow.

Signed-off-by: Katarzyna Giądła <katarzyna.giadla@nordicsemi.no>

jfischer-no committed:
samples: bluetooth: remote_shell: use CDC ACM serial backend

Use CDC ACM serial backend from the new USB device support.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
samples: peripheral: use CDC ACM serial backend

Use CDC ACM serial backend from the new USB device support.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

jfischer-no committed:
samples: wifi: allow samples using legacy USB stacks to pass CI

Add CONFIG_DEPRECATION_TEST=y to sample configuration.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

tokangas committed:
applications: serial_lte_modem: Fix compilation warning

Fixed a compilation warning from configure_power_pin_interrupt().

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Fix MCUboot serial CDC ACM CI failures

The MCUboot serial recovery over USB CDC ACM still uses USB legacy
stack. Disable deprecation warnings to workaround build failures in CI.

Jira: NCSDK-35440

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

gWacey committed:
applications: nrf5340_audio: Updates for upmerge

As using the deprecated USB stack, must turn off
deprecation warnings with CONFIG_DEPRECATION_TEST=y.

Change to the endpoint structure.

Signed-off-by: Graham Wacey <graham.wacey@nordicsemi.no>

ppryga-nordic committed:
tests: bluetooth: iso Remove uses of auto name from advertising params

The BT_LE_ADV_OPT_USE_NAME option was deprecated. There is no
automatic add of a name to advertising params. It must be done
manually by call to bt_le_ext_adv_set_data() in case of extended
advertising.

The commit changes the ISO test code that was still using the
deprecated option.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Fix legacy USB CI failures

The nRF Desktop still uses USB legacy stack in some configurations.
Disable deprecation warnings to workaround build failures in CI.

Jira: NCSDK-35440

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

jfischer-no committed:
applications: connectivity_bridge: allow application to pass CI

Add CONFIG_DEPRECATION_TEST=y to sample configuration.
Application uses deprecated code and needs to be ported to the new USB
device stack.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

seov-nordic committed:
samples: pmic: fix devicetree errors on uicr build

By applying shields only to the main image

Signed-off-by: Sergei Ovchinnikov <sergei.ovchinnikov@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Suppress deprecation warnings in usb_state.c

Change suppresses build warnings related to deprecated APIs of USB
legacy stack if USB next stack is chosen.

Jira: NCSDK-35440

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
tests: fast_pair: crypto: Drop unit tests for TinyCrypt

TinyCrypt library is no longer supported by Zephyr.

Jira: NCSDK-35445

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

MarekPieta committed:
bluetooth: fast_pair: Drop TinyCrypt Fast Pair crypto backend

TinyCrypt library is no longer supported by Zephyr.

Jira: NCSDK-35445

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

jfischer-no committed:
tests: mcuboot: external_flash: allow test to pass CI

Add CONFIG_DEPRECATION_TEST=y to test configuration.

Signed-off-by: Johann Fischer <johann.fischer@nordicsemi.no>

michalek-no committed:
sysbuild: kconfig: MCUBoot default swap mode

sets MODE_SWAP_USING_MOVE as an default for NORDIC chips

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

doki-nordic committed:
samples: nfc: writable_ndef_msg: Select config CRC when ZMS is enabled

The sample in Kconfig sets default ZMS to 'y', but ZMS also must select
CRC. It is done in different place, but because how Kconfig works
it must be also set here.

Jira: NCSDK-35447
Jira: NCSDK-35446
Signed-off-by: Dominik Kilian <dominik.kilian@nordicsemi.no>

MarekPieta commented on samples/nfc/writable_ndef_msg/Kconfig:
```diff
@@ -7,6 +7,7 @@
 source "Kconfig.zephyr"
 
 config ZMS
+	select CRC
```
Shouldn't we enable `CONFIG_FLASH_MAP` in configuration instead of adding the `select CRC` here?

MarekPieta replied:
IIRC `source "Kconfig.zephyr"` should go below sample-specific default values overlays too

doki-nordic replied:
Done

ArekBalysNordic committed:
samples: openthread: Drop support for L2 in coprocessor

L2 variant is not needed in coprocessor sample. We can use there
only the raw nrf-radio version.

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>

carlescufi committed:
doc: Add missing dependency: sphinxcontrib.programoutput

Add this missing dependency which is used in the Zephyr doc build.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

grochu committed:
subsys: nfc: add explicit cast in LE OOB record encoder for LE Role enum

On LLVM toolchain, the lack of cast produces an error of incompatible
pointer types.

Ref. NCSDK-35444

Signed-off-by: Michał Grochala <michal.grochala@nordicsemi.no>

jonathannilsen committed:
sysbuild: remove PERIPHCONF generation via nrf-regtool uicr-migrate

PERIPHCONF generation based on devicetree is being added in a script
in upstream zephyr. The new script script replaces nrf-regtool and the
temporary "periphconf migration" solution.

Ref: NCSDK-35206

Signed-off-by: Jonathan Nilsen <jonathan.nilsen@nordicsemi.no>

grochu committed:
samples: bluetooth: peripheral_status: - remove thingy53 support

Temporarily remove Thingy53 support as it doesn't build after the
USB stack change in Zephyr.
TODO: migrate to the new USB stack.

Signed-off-by: Michał Grochala <michal.grochala@nordicsemi.no>

doki-nordic committed:
samples: nfc: writable_ndef_msg: Select FLASH_MAP when ZMS is enabled

The sample in Kconfig sets default ZMS to 'y', but CONFIG_FLASH_MAP
must also be set in this case.

Jira: NCSDK-35447
Jira: NCSDK-35446

Signed-off-by: Dominik Kilian <dominik.kilian@nordicsemi.no>

MarekPieta commented on samples/nfc/writable_ndef_msg/Kconfig:
```diff
@@ -7,7 +7,9 @@
 source "Kconfig.zephyr"
```
`source "Kconfig.zephyr"` should be below sample-specific overlays of defaults to ensure that the sample-specific overlays of Kconfig defaults would be taken into account
